cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/model/User.java

```diff
@@ -1,78 +1,147 @@
+    public static final String CUSTOM_ATTRIBUTE_IDENTIFIERS = "custom:identifiers";
+    public static final String CUSTOM_ATTRIBUTE_CUSTOMER_ID = "custom:customerId";
+    public static final String CUSTOM_ATTRIBUTE_CRISTIN_ID = "custom:cristinId";
+    private static final Logger logger = LoggerFactory.getLogger(User.class);
+    private final UserAttributes userAttributes;
+    private final boolean newUser;
+
+    private String userPoolId;
+    private String cognitoUsername;
+
+    private UserDto apiUser;
+    private final UserService userService;
+
+    public User(String userPoolId,
+                String cognitoUsername,
+                UserDto apiUser,
+                UserAttributes userAttributes,
+                boolean isNewUser,
+                UserService userService) {
+        this.userPoolId = userPoolId;
+        this.cognitoUsername = cognitoUsername;
+        this.apiUser = apiUser;
+        this.userAttributes = userAttributes;
+        this.newUser = isNewUser;
+        this.userService = userService;
+    }
+
+    public String getUserPoolId() {
+        return userPoolId;
     }
 
-    public String getUsername() {
-        return username;
+    public void setUserPoolId(String userPoolId) {
+        this.userPoolId = userPoolId;
     }
 
-    public String getGivenName() {
-        return givenName;
+    public String getCognitoUsername() {
+        return cognitoUsername;
     }
 
-    public String getFamilyName() {
-        return familyName;
+    public void setCognitoUsername(String cognitoUsername) {
+        this.cognitoUsername = cognitoUsername;
     }
 
-    public String getInstitution() {
-        return institution;
+    public UserDto getApiUser() {
+        return apiUser;
     }
 
-    public List<Role> getRoles() {
-        return roles;
+    public void setApiUser(UserDto apiUser) {
+        this.apiUser = apiUser;
     }
 
-    @Override
-    @JacocoGenerated
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
+    public boolean hasCustomerAttributes() {
+        return getApiUser().getCristinId() != null && getApiUser().getInstitution() != null;
+    }
+
+    /*public void updateCustomAttributesInUserPool() {
+        userService.updateUserAttributes(
+            userPoolId,
+            cognitoUsername,
+            getAttributeTypesToUpdate(userAttributes));
+    }*/
+
+    public void notifyUserManagementIfNewUser() {
+        // TODO: Send aync message to usermgmt
+        // so user mgmt can pick up and run API calls for updating:
+        // - Cognito User pool (authorative idp)
+        // - Dynamodb UsersAndRoles table.  (authorative users)
+
+        // Now temporary fire a thread with fire and forget ..
+        // TODO Remove this! Fix implementation in UserManagement.
+        // No gurantee this code finishes!
+        if (newUser) {
+            new Thread(() -> {
```

Comment:
Hack.. task says to use EventBridge (which is what we ideally want). 

But it requires to setup EventBridge, DLQ, SQS and make publishers. Is that a lot of work? We probably want to do this .. 

@janbj